presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -886,6 +890,26 @@ private void invalidatePartitionCache(String databaseName, String tableName)
         return get(userTablePrivileges, new UserTableKey(user, tableName, databaseName));
     }
 
+    @Override
+    public void grantTablePrivilege(String databaseName, String tableName, Identity identity, PrivilegeGrantInfo privilegeGrantInfo)
+    {
+        try (HiveMetastoreClient metastoreClient = clientProvider.createMetastoreClient()) {
+            HiveObjectPrivilege privilegeObject = new HiveObjectPrivilege(
+                    new HiveObjectRef(HiveObjectType.TABLE, databaseName, tableName, null, null),
+                    identity.getUser(),
+                    PrincipalType.USER, // TODO: check whether it is USER or ROLE.
+                    privilegeGrantInfo);
+            ImmutableList.Builder<HiveObjectPrivilege> privilegeBagList = ImmutableList.builder();
+            privilegeBagList.add(privilegeObject);
+
+            // Should you check whether the user already has the given privilege and if yes, return.
```

Comment:
It's a todo. I am not entirely sure whether it is worth checking the privileges. I added the comment, just to keep that in mind.